packages/server/src/course/course.controller.ts

```diff
@@ -893,4 +896,84 @@ export class CourseController {
       });
     return;
   }
+
+  // UPDATE course_settings_model SET selectedFeature = false WHERE courseId = selectedCourseId;
+  // will also create a new course settings record if it doesn't exist for the course
+  @Patch(':id/features')
+  @UseGuards(JwtAuthGuard, CourseRolesGuard)
+  @Roles(Role.PROFESSOR)
+  async enableDisableFeature(
+    @Param('id') courseId: number,
+    @Body() body: CourseSettingsRequestBody,
+  ): Promise<void> {
+    // fetch existing course settings
+    let courseSettings = await CourseSettingsModel.findOne({
+      where: { courseId: courseId },
+    });
+
+    // if no course settings exist yet, create new course settings for the course
+    if (!courseSettings) {
```

Comment:
I feel like they should be created when course is created, this controller endpoint shouldn't be responsible for doing this job?

Reply:
I had also thought about this, and I mostly just did this for backwards compatibility